guidance/src/guidance/guidance_worker.cpp

```diff
@@ -157,6 +157,16 @@ namespace guidance
         }
     }
 
+    void GuidanceWorker::create_guidance_state_machine()
+    {
+        try {
+            guidance_state_machine_factory.createStateMachineInstance(vehicle_state_machine_type);
```

Comment:
Looks like you forgot to assign the returned value here. 
Also I guess since you are returning the nullptr this try catch won't notice it. It would be better to have an if statment that checks for the nullptr and if it occurs then it calls nh_.handleException() and passes in an std::invalid_argument exception containing a log describing that an invalid parameter was set. This will cause the log to made and the node to shutdown cleanly.